src/main/scala/guessbook/GuessBookUserInput.scala

```diff
@@ -0,0 +1,206 @@
+//    val selectedBooks = List(game.selectBook(game.allBooks), game.selectBook(game.allBooks))
+
+    // Each player selects the book that the other one must guess
+    val selectedBookP2: Book = selectBookForOpponent(1)
+    val selectedBookP1: Book = selectBookForOpponent(2)
+    val selectedBooks = List(selectedBookP1, selectedBookP2)
+
+    val booksOnBoard = Array(game.allBooks, game.allBooks)
+    var gameOver = false
+    var guesser = 2
+
+    while (!gameOver) {
+      guesser = guesser % 2 + 1 // 1 -> 2, 2 -> 1
+      println(s"\nPlayer $guesser's turn!")
+      println(s"Books remaining on P$guesser's board:")
+      game.printRemainingBooks(booksOnBoard(guesser-1), maxBooksPerLine = 5)
+
+      val guessType: Int = chooseAttributeOrTitle()
+      guessType match {
+        case 1 => { // guessing attribute and value
+          val guessAttribute: BookAttribute = specifyGuessAttribute()
+          guessAttribute match {
+            case Honkaku | SchoolOrUniversityStudents | Series | PushkinVertigo => {
+              val guessValue: Boolean = specifyBooleanValue()
+              booksOnBoard(guesser-1) = game.poseQuestion(selectedBooks(guesser-1), guessAttribute, guessValue, booksOnBoard(guesser-1))
+            }
+            case AuthorGender => {
+              val guessValue: GenderEnum.Value = specifyEnumValue(GenderEnum)
+              booksOnBoard(guesser-1) = game.poseQuestion(selectedBooks(guesser-1), guessAttribute, guessValue, booksOnBoard(guesser-1))
+            }
+            case CountryOfOrigin => {
+              val guessValue: Country.Value = specifyEnumValue(Country)
+              booksOnBoard(guesser-1) = game.poseQuestion(selectedBooks(guesser-1), guessAttribute, guessValue, booksOnBoard(guesser-1))
+            }
+            case DetectiveType => {
+              val guessValue: Detective.Value = specifyEnumValue(Detective)
+              booksOnBoard(guesser-1) = game.poseQuestion(selectedBooks(guesser-1), guessAttribute, guessValue, booksOnBoard(guesser-1))
+            }
+            case SuspectPoolType => {
+              val guessValue: SuspectPool.Value = specifyEnumValue(SuspectPool)
+              booksOnBoard(guesser-1) = game.poseQuestion(selectedBooks(guesser-1), guessAttribute, guessValue, booksOnBoard(guesser-1))
+            }
+          }
+
+          println(s"Books remaining on P$guesser's board:")
+          game.printRemainingBooks(booksOnBoard(guesser-1), maxBooksPerLine = 5)
+        }
+        case 2 => { // guessing book title
+          val guessedTitle: String = specifyBookTitle()
+          val guessResult = game.guessBookTitle(selectedBooks(guesser-1), guessedTitle, guesser, booksOnBoard(guesser-1))
+          booksOnBoard(guesser-1) = guessResult._1
+          gameOver = guessResult._2
+          if (!gameOver){
+            println(s"Books remaining on P$guesser's board:")
+            game.printRemainingBooks(booksOnBoard(guesser-1), maxBooksPerLine = 5)
```

Comment:
is maxBooksPerLine used to emulate a guess who (or guess what in this case :D ) board?

Reply:
It does have that effect! But I added it so that the list of books wouldn't go off the screen

Reply:
In that case, would it make sense to limit it by total characters rather than number of books?

Reply:
Yes, that would make the lines more even... although using number of books could help the player to easily count how many books are left on the board. Still, I've created a new version of the function with maxCharsPerLine